topics/dev/tutorials/debugging/tutorial.md

```diff
@@ -0,0 +1,730 @@
+>    >  
+>    >  The solution is hidden below, but try your hand at fixing it first:
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failure is only happening on the disabled test case, so take a look in the template of `SavedRulesSelector` where the conditions for adding disabled are defined.  
+>    >  > The way it's worded in the test (and the way one could think about it logically), is that if there is no history, the button should be disabled, but the code says 
+>    >  > ```javascript
+>    >  > :class="{ disabled: numOfSavedRules == 1 }"
+>    >  > ``` 
+>    >  > -- whoops! A very silly off by one error. On that very same line, change the 1 to a 0, and run the test again. 
+>    >  {: .solution }
+>    >  
+>    >  If you think you've got it, try running the test locally again to be sure.
+>    >  
+>    >  > ### {% icon comment %} Running fewer client tests
+>    >  >
+>    >  > If you don't want to run the whole suite of client tests, you can add keywords that match the test path/name. For example both `yarn run jest rule` and `yarn jest selector` will work, but the former will run other tests with rule in the path/module name.  
+>    >  {: .comment}
+>    >  
+> 4. __Finishing up__
+>    >  
+>    >  Nice Job! If you're here, you've found the problem, ran it locally, fixed the problem, ran it locally again without errors, and are ready to commit your change. Write a useful commit message like `Fix logic error in SavedRuleSelector` or something similar, making sure it is clear what you've done.
+>    > (See [Galaxy's contributor's guide](https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTING.md#how-to-contribute) for details on how to write a useful and properly formatted commit message.)
+>    >  
+>    >  Awesome work - you now understand how to solve client linting issues!- you now have a handle on client tests!
+{: .hands_on}
+
+# Selenium test failure
+
+Selenium is the end-to-end or integration testing framework that we use; so these tests often are designed to test both frontend and backend code. This makes fixing them a little difficult, but we are sure you can do it!
+
+> ### {% icon hands_on %} Hands-on: Fixing a failing Selenium test
+> 
+> 1. __Finding the failing test on GitHub__
+>    >  
+>    >  There may be actually a few failing Selenium tests, but the good new is, they are the result of only one little error. The failing tests on GitHub says "Selenium Test 3.7, 2". When you click on the details of this one, it's not always very clear which test is causing the failure, but if you download the artifacts, it's very clear which tests have failed.
+>    >  
+>    >  Try to identify which tests are failing based on artifacts you downloaded.
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failing tests are:
+>    >  >
+>    >  > - `test_build_list_and_show_items`
+>    >  > - `test_data_column_input`
```

Comment:
@ic4f This test is not failing... whoops? We should remove this line

Reply:
This has been updated. I'll edit the wording too to make it clear that more than one workflow needs to be checked.